source/blender/nodes/shader/nodes/node_shader_bsdf_principled.c

```diff
@@ -78,7 +89,86 @@ static int node_shader_gpu_bsdf_principled(GPUMaterial *mat, bNode *UNUSED(node)
 	else
 		GPU_link(mat, "direction_transform_m4v3", in[18].link, GPU_builtin(GPU_VIEW_MATRIX), &in[18].link);
 
-	return GPU_stack_link(mat, "node_bsdf_principled", in, out, GPU_builtin(GPU_VIEW_POSITION));
+
+	// Init values
+	GPU_link(mat, "node_bsdf_principled_summation_init", &summation);
+	GPU_stack_link(mat, "node_bsdf_principled_adquired_in", in, out, &in0, &in1, &in2, &in3, &in4, &in5,
+	               &in6, &in7, &in8, &in9, &in10, &in11, &in12, &in13, &in14, &in15, &in16, &in17, &in18,
+	               &in19); 
+
+	sce = GPU_material_scene(mat);
+	world = sce->world;
+	world_envlight_energy = GPU_envlight_energy();
+	world_envlight_linfac = GPU_envlight_linfac();
+	world_envlight_logfac = GPU_envlight_logfac();
+
+	// Recursive BSDF shading through all lamps
+	for (SETLOOPER(sce, sce_iter, base)) {
+		Object *ob = base->object;
+
+		if (ob->type == OB_LAMP) {
+			GPULamp *lamp = GPU_lamp_from_blender(sce, ob, NULL);
+			if (lamp) {
+				visifac = GPU_lamp_get_data(mat, lamp, &col, &lv, &dist, &shadow, &energy);
+
+				GPU_link(mat, "node_bsdf_principled", in0, in1, in2, in3, in4, in5, in6, in7, 
+				         in8, in9, in10, in11, in12, in13, in14, in15, in16, in17, in18, in19, GPU_builtin(GPU_VIEW_POSITION),
+				         col, energy, lv, visifac, &summation_partial);
+				GPU_link(mat, "node_bsdf_principled_add", summation_partial, summation, &summation);
+
+			}
+		}
+	}
+	
+	// Ambient lighting and color
+	if (world) {
+		// exposure correction
+		if (world->exp != 0.0f || world->range != 1.0f) {
```

Comment:
These checks are fine when the user is using "constant values" for his material (= no uniforms) else the user could set a 0 exposure and range and modify it later.

To avoid such issue you can mimic similar check in gpu_material.c:
```cpp
if (world->exp != 0.0f || world->range != 1.0f || !(ma->constflag & MA_CONSTANT_WORLD)) {
```